### ISSUE TITLE
Add support for consistency regularization

### ISSUE THREAD

**Issue by IgorSusmelj** [lightly-ai/lightly#126]
Found a very interesting paper to address the problem that negatives could be similar to the positives (e.g. same class). They introduce a rather simple regularization term for contrastive methods. 

Here is the paper: [CO2: CONSISTENT CONTRAST FOR UNSUPERVISED VISUAL REPRESENTATION LEARNING](https://arxiv.org/pdf/2010.02217.pdf)

We could maybe even use a similar implementation as here https://github.com/perrying/pytorch-consistency-regularization/tree/master/ssl_lib/consistency

I would propose to add this as a new loss function.
